biz.aQute.bndlib/src/aQute/bnd/service/reporter/ReportGeneratorService.java

```diff
@@ -0,0 +1,65 @@
+package aQute.bnd.service.reporter;
+
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This plugin can generate a DTO report of an arbitrary object. One can
+ * implements the {@link ReportEntryPlugin} plugin for a specific object type to
+ * contributed to a report.
+ */
+@ProviderType
+public interface ReportGeneratorService {
+
+	/**
+	 * Generate a DTO report of the source object.
+	 * 
+	 * @param source the source object from which the report is generated, if
+	 *            {@code null} an empty report is returned.
+	 * @return the report, never {@code null}
+	 */
+	public Map<String, Object> generateReportOf(final Object source);
+
+	/**
+	 * Generate a DTO report of the source object.
+	 * 
+	 * @param source the source object from which the report is generated, if
+	 *            {@code null} an empty report is returned.
+	 * @param filter a filter used to filter which {@link ReportEntryPlugin}
+	 *            plugins will contribute to the report, if not specified all
+	 *            configured {@link ReportEntryPlugin} plugins that handle the
+	 *            source object type will contribute.
+	 * @return the report, never {@code null}
+	 */
+	public Map<String, Object> generateReportOf(final Object source, final String filter);
```

Comment:
Is this an OSGi LDAP filter expression?

Reply:
I see from looking at your impl branch that it is. The javadoc here should be more clear.